task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -726,7 +727,12 @@ def _handle_request(self, msg: ToSupervisor, log: FilteringBoundLogger):
             resp = var_result.model_dump_json().encode()
         elif isinstance(msg, GetXCom):
             xcom = self.client.xcoms.get(msg.dag_id, msg.run_id, msg.task_id, msg.key, msg.map_index)
-            xcom_result = XComResult.from_xcom_response(xcom)
+            if isinstance(xcom, XComResponse):
+                xcom_result = XComResult.from_xcom_response(xcom)
+            else:
+                # Airflow 2.x just ignores the absence of an XCom and moves on with a return value of None
+                # Hence making this resp with key as `key` and value as None, so that the message is sent back to task runner.
+                xcom_result = XComResult.from_xcom_response(XComResponse(key=msg.key, value=None))
```

Comment:
This branch indicates that the "xcom" is not found (The server responded 404).
Instead of complicating the client which is the sdk, I chose to send a message to the task runner (it will receive it through get_message) with an empty "value". If the value is empty, the `xcom_pull` in task runner will return the `default` which is `None` anyways: https://github.com/apache/airflow/blob/main/task_sdk/src/airflow/sdk/execution_time/task_runner.py#L198-L204 which is the behaviour I am intending.

Reply:
Why are you not "just" moving the fallback handling into task_sdk/src/airflow/sdk/api/client.py but return an ErrorReponse which then later needs to be converted? Is this not over-complicating?

Reply:
I briefly gave it some thought and thought that the client shouldn't have to handle such a scenario. But now that I think about it, that will significantly simplify it

Reply:
Let me actually make that change

Reply:
I actually handled it that way and its much simpler and cleaner.